### PR TITLE
[CLI-1031] Added setTimeout, so the node processes will be able to ex…

### PR DESCRIPTION
**Pull request by vladimir-trifonov**
…it, when had pressed ctrl+c on windows

https://jira.appcelerator.org/browse/CLI-1031

### CODE REVIEW

File: lib/run.js [hunk 1]
```diff
@@ -82,13 +82,17 @@ function run(installBin, additionalArgs, cb, dontexit) {
...
+					// process to remain alive, mainly because the child wasn't able to leave on time
+					(setTimeout(function () {
+						debug('signal received', name, 'sending to', child);
+						if (name === 'exit') {
+							child.kill();
+							process.exit(arguments[0]);
+						} else {
+							child.kill(name);
+						}
+					}, 500)).unref();
```
Comment (janvennemann) [inline lib/run.js]: I think we have a similar solution somewhere else, but solving this with a timeout it more of a workaround. This happens due to Windows lacking the support to send signals. Sending a `SIGINT` (generated when pressing CTRL+C) to a child will cause the unconditional termination of the target process. This is why any listeners in the child process won't be called and we end up with an orphaned process. 
I'd propose to include a check wether we are NOT on Windows and only then send any signals to the child. Windows will handle the child termination itself and send the appropriate signal to all childs which then in turn are able to run their signal handlers.

Comment (vladimir-trifonov) [inline lib/run.js]: Make sense, I gonna try it tomorrow.

Comment (vladimir-trifonov) [inline lib/run.js]: Did you mean to prevent sending only SIGINT, or the other signals also on win?

**Comment by vladimir-trifonov** [comment]
Done!

**Comment by vladimir-trifonov** [comment]
Done!

**Comment by alasdairhurst** [comment]
Works perfectly for me. @janvennemann notice any other glaring issues? I've been cleaning up the arrow unit tests to work on windows and this is pretty much a requirement.

**Comment by janvennemann** [comment]
@vladimir-trifonov @alasdairhurst, works for me too! Approved!

**Comment by vladimir-trifonov** [comment]
Done!

**Pull request merged**